### PR TITLE
SS-7557 Finalize the first App Builder template

### PR DESCRIPTION
**Pull request by snabela**

### CODE REVIEW

File: src/hooks/ui/useCustomTheme.ts [hunk 1]
```diff
@@ -235,6 +236,25 @@ export const useCustomTheme = () => {
...
+						components: {
+							Paper: {
+								defaultProps: {
+									withBorder: false,
+								}
+							}
+						}
+					}
+				}
+			}),
```
Comment (snabela) [inline src/hooks/ui/useCustomTheme.ts:257]: @mathieu-shapediver here one can now define theme overrides per container

File: src/hooks/ui/useCustomTheme.ts [hunk 2]
```diff
@@ -183,6 +185,41 @@ export const useCustomTheme = () => {
...
+						}
+					},
+					/** Theme overrides for the "bottom" container. */
+					bottom: {
+						components: {
+							AppBuilderTextWidgetComponent: AppBuilderTextWidgetThemeProps({
+								styles: { root: { overflow: "auto" }}
+							}),
+						}
+					}
```
Comment (snabela) [inline src/hooks/ui/useCustomTheme.ts:220]: @mathieu-shapediver in the bottom container, we use scrollbars for the text widgets

File: src/pages/templates/AppBuilderContainerWrapper.tsx
Comment (snabela) [inline src/pages/templates/AppBuilderContainerWrapper.tsx:1]: Wrapper for all AppBuilder containers, providing context to the nodes below it.

**Comment by snabela** [comment]
@mathieu-shapediver please test here: https://appbuilder.shapediver.com/v1/main/0.3.0-beta.4/?g=iframe9.json
Note that [iframe9.json](https://appbuilder.shapediver.com/v1/main/0.3.0-beta.4/iframe9.json) overrides the AppBuilder output of the model, because there is a bug in the image widget component in GH which I still need to fix.

**Review by mathieu-shapediver — CHANGES_REQUESTED** [review]
@snabela It seems really f**** stunning!
I would suggest the following things:
* The navbar on the left should be wider. I would go for navbarWidth: { md: 250, lg: 300 } .
* I would set all anchors within text widgets with "target=_blank", which will be crucial for the header bar in particular but in general preferable for all anchors within apps.
* I would add the possibility to override some styling properties of anchors in the template, in particular the text color and the underline property. Specifically for the top header bar: I think the underline property should be set to "never" and by default the color to black.

**Comment by mathieu-shapediver** [comment]
@snabela Additional thing: I noticed that the left navigation bar has a vertical separation line on the right, but the right container does not have one on the left. We should also have this line for the right container.

**Comment by snabela** [comment]
> * The navbar on the left should be wider. I would go for navbarWidth: { md: 250, lg: 300 } .

done here: https://github.com/shapediver/ShapeDiverReactExample/pull/71/commits/47743f282a415265e91246104cfc74fe72f63029

> * I would set all anchors within text widgets with "target=_blank", which will be crucial for the header bar in particular but in general preferable for all anchors within apps.
> * I would add the possibility to override some styling properties of anchors in the template, in particular the text color and the underline property. Specifically for the top header bar: I think by default the color should be set to var(--text-color).

done here: https://github.com/shapediver/ShapeDiverReactExample/pull/71/commits/ceaeb6e09e6705c73bf412d9b847ffd5d734aeaf

The color of the anchor is now inherited from the parent by default (this behavior can be controlled by the theme, see the commit). This means you can now color anchors like this in markdown: `:span[[LinkedIn](https://www.linkedin.com/company/intersticestudio/)]{color="red"}`

**Comment by snabela** [comment]
> @snabela Additional thing: I noticed that the left navigation bar has a vertical separation line on the right, but the right container does not have one on the left. We should also have this line for the right container.

done here: https://github.com/shapediver/ShapeDiverReactExample/pull/71/commits/0adce67aafb44be7374761652be95d4122896ab4

**Review by mathieu-shapediver — APPROVED** [review]

**Pull request merged**